Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/FirePropagationSystem.cs

```diff
@@ -30,44 +30,100 @@ protected override void OnUpdate()
 
-        Job.WithCode(() => {
-
-            for (var i = 0; i < heatmapBuffer.Length; i++)
-            {
-                if (heatmapBuffer[i] >= HEAT_THRESHOLD)
-                {
-                    heatmapBuffer[i] += deltaTime;
-                    HeatAdjacents(ref heatmapBuffer, localCheckAdjacents, i, heatmapData.mapSideLength, deltaTime);
-                }
-
-                if (heatmapBuffer[i] >= 1f)
-                    heatmapBuffer[i] = 1f;
-            }
+        var job = new HeatJob()
+        {
+            width = heatmapData.mapSideLength,
+            adjacentOffsets = localCheckAdjacents,
+            deltaTime = deltaTime,
+            heatmapBuffer = heatmapBuffer
+        };
 
-        }).Run();
+        var handle = job.Schedule(heatmapBuffer.Length, 32);
+        handle.Complete();
+        
     }
 
     static void HeatAdjacents(
         ref DynamicBuffer<HeatMapTemperature> buffer, 
         NativeArray<int2> adjacentOffsets, 
         int tileIndex,  int width, float deltaTime)
     {
+#if true
         for (int iCheck = 0; iCheck < adjacentOffsets.Length; iCheck++)
         {
-            int2 tileCoord = GridUtil.GetTileCoordinate(tileIndex, width);
+            int2 tileCoord = GridUtility.GetTileCoordinate(tileIndex, width);
             int x = tileCoord.x + adjacentOffsets[iCheck].x;
             int z = tileCoord.y + adjacentOffsets[iCheck].y;
 
             bool inBounds = (x >= 0 && x <= width-1 && z >= 0 && z <= width-1);
 
             if (inBounds)
             {
-                int adjacentIndex = GridUtil.GetTileIndex(x, z, width);
+                int adjacentIndex = GridUtility.GetTileIndex(x, z, width);
                 
                 if(buffer[adjacentIndex] < HEAT_THRESHOLD)
                     buffer[adjacentIndex] += deltaTime;
             }
         }
+#else
+        var job = new HeatAdjacentsJob()
+        {
+            width = width,
+            tileIndex = tileIndex,
+            adjacentOffsets = adjacentOffsets,
+            adjacentIndexArray = adjacentIndices
+        };
+
+        var handle = job.Schedule(adjacentOffsets.Length, 1);
+        handle.Complete();
+
+        for (int iCheck = 0; iCheck < adjacentIndices.Length; iCheck++)
+        {
+            int adjacentIndex = adjacentIndices[iCheck];
+            if( adjacentIndex >= 0 && buffer[adjacentIndex] < HEAT_THRESHOLD)
+                buffer[adjacentIndex] += deltaTime;
+        }
+#endif
+    }
+    
+    [BurstCompile]
+    struct HeatJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<int2> adjacentOffsets;
+        
+        [NativeDisableParallelForRestriction]
+        public DynamicBuffer<HeatMapTemperature> heatmapBuffer;
+        
+        public float deltaTime;
+        public int width;
+        public void Execute(int tileIndex)
+        {
+            if (heatmapBuffer[tileIndex] >= HEAT_THRESHOLD)
+            {
+                heatmapBuffer[tileIndex] += deltaTime;
```

Comment:
heat growth could use a constant; same with heat propagation.  fire seems to spread too quick.